### ISSUE TITLE
Make tuning models easier: enable setting learning rate in sqlflow

### ISSUE THREAD

**Issue by shendiaomo** [sql-machine-learning/sqlflow#869]
**Is your feature request related to a problem? Please describe.**
In this moment, we cannot set learning rate for DNNs or other models, which makes tuning models directly in sqlflow very hard.
**Describe the solution you'd like**
Support setting learning rate in the `TRAIN' clause.

**Describe alternatives (optional)**
Tuning models in raw tf/xgboost/keras scripts and filling the optimal learning rate into sqlflow premade model source files.

**Additional Notes**
_Pros_: It's easier for algorithm engineers to tune models.
_Cons_: The sqlflow syntax may become a bit complex.
The feature can be implemented in one of the two forms as below:
_1. TRAIN Classifier with **model**.learning_rate = 0.01
2. TRAIN Classifier with **optimizer**.learning_rate = 0.01_
The latter is preferred because we may need to set more optimizer hyperparameters in the future.

**Comment by Yancey0623** [comment]
I agree with the second one
> 2. TRAIN Classifier with optimizer.learning_rate = 0.0

All the attributes with the prefix `optimizer.` can be the arguments of Optimizer API, and using 
```text
train.optimizer = "Adam"
```

to specify which optimizer algorithm would be used.

**Comment by typhoonzero** [comment]
I think this issue is not only about `learning_rate`, as described here: https://www.tensorflow.org/api_docs/python/tf/estimator/DNNClassifier, we can configure the optimizer like :

```python
estimator = DNNClassifier(
    feature_columns=[categorical_feature_a_emb, categorical_feature_b_emb],
    hidden_units=[1024, 512, 256],
    optimizer=lambda: tf.AdamOptimizer(
        learning_rate=tf.exponential_decay(
            learning_rate=0.1,
            global_step=tf.get_global_step(),
            decay_steps=10000,
            decay_rate=0.96))
```

So all `optimizer.xxx` parameters should go into this `optimizer=...` parameter. Additionally we also need a attribute `optimizer.type` which is the optimizer name supported by Tensorflow.

**Comment by typhoonzero** [comment]
A quick example SQL should be like:

```sql
SELECT * from iris.train
TRAIN DNNClassifier
WITH model.n_classes=3,
          optimizer.type="Adam",
          optimizer.learning_rate=0.1,
          optimizer.learning_rate_decay="exponential",
          optimizer.decay_steps=10000,
          optimizer.decay_rate=0.96
COLUMN ...
INTO modeldb.mymodel;
```

**Comment by typhoonzero** [comment]
Fixed by #1376
